### PR TITLE
OOP library

### PR DESCRIPTION
**Pull request by shella12**
# OOP school library: add Person, Student and Teacher classes

- Created class Person with the following:

      - Instance vars: @id, @name, and @age.
      - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
      - Getters for @id, @name, and @age.
      - Setters for @name and @age.
      - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
      - Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Created class Student with the following:

      - Inherits from Person.
      - Constructor extends parent's constructor by adding @classroom and a parameter for it.
      - Method play_hooky that returns "¯\(ツ)/¯".

- Created class Teacher with the following:

      - Inherits from Person.
      - Constructor extends parent's constructor by adding @specialization and a parameter for it.
      - Override can_use_services? so it always returns true.
      - Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    is_of_age? or @parent_permission
+  end
+
+  private
+
+  def of_age?
```
Comment (pinkmoon25) [inline person.rb:18]: - [ ] I think you accidentally misplaced the name of this method, this should be `is_of_age?` which you have referenced in your public method too, please fix this :+1:

File: README.md [hunk 2]
```diff
@@ -48,48 +12,26 @@ After you're finished please remove all the comments and instructions!
...
-
-<details>
-  <summary>Client</summary>
-  <ul>
-    <li><a href="https://reactjs.org/">React.js</a></li>
-  </ul>
-</details>
-
 <details>
   <summary>Server</summary>
```
Comment (pinkmoon25) [inline README.md]: - [ ] [OPTIONAL] Ruby is not a server, I suggest you fix this on your Readme :+1: 
image

File: README.md [hunk 3]
```diff
@@ -99,123 +41,82 @@ After you're finished please remove all the comments and instructions!
 
 > Describe between 1-3 key features of the application.
```
Comment (pinkmoon25) [inline README.md]: - [ ] Please remove all these placeholder texts from your Readme , they are there to help you fill in the details and should not be present on your committed or professional readme version :+1:

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @shella12 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
linters :heavy_check_mark: 
gitflow :heavy_check_mark: 
ruby classes :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @shella12,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**